app/controllers/feedbacks_controller.rb

```diff
@@ -1,11 +1,28 @@
 class FeedbacksController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_lecture
+  before_action :validate_joined_user_or_owner
+
   def create
-    @lecture = Lecture.find(params[:lecture_id])
     @feedback = @lecture.feedbacks.create(comment_params)
   end
 
   private
     def comment_params
       params.require(:feedback).permit(:content)
     end
+
+    def validate_joined_user_or_owner
+      isStudent = current_user.is_student
+      isJoinedStudent = isStudent && @lecture.participating_students.include?(current_user)
+      isLectureOwner = !isStudent && @lecture.lecturer == current_user
+      # return head :forbidden unless isJoinedStudent || isLectureOwner
+      unless isJoinedStudent || isLectureOwner
+        redirect_to course_path(@lecture.course), alert: "You are not a member of this lecture!"
```

Comment:
Does this work for the feedback? Do you not need to handle this like the questions?

Reply:
Check this: works fine :+1: